applications: nrf5340_audio: doc: Remove buildprog from DFU chapter (#14843)
nrfconnect/sdk-nrf: applications/nrf5340_audio/doc/fota.rst

   :local:
   :depth: 2

The nRF5340 Audio applications all support FOTA upgrades, and the application implementation is based on the procedure described in :ref:`ug_nrf53_developing_ble_fota`.

Requirements for FOTA
*********************

> If the application is running on the nRF5340 Audio DK an external flash shield is needed in order to update both the application and network core at the same time.

Reviewer: Let's keep the tone active.
```suggestion
If the application is running on the nRF5340 Audio DK, you need an external flash shield to upgrade both the application and network core at the same time.
```

To test Firmware Over-The-Air (FOTA), you need an Android or iOS device with the `nRF Connect Device Manager`_ app installed.

To enable the external flash DFU and do FOTA upgrades for the application core and the network core at the same time, you need an external flash shield.
If the application is running on the nRF5340 Audio DK an external flash shield is needed in order to update both the application and network core at the same time.
See `Requirements for external flash memory DFU`_ in the nRF5340 Audio DK Hardware documentation for more information.

.. _nrf53_audio_app_configuration_configure_fota:

Configuring FOTA upgrades
*************************

.. caution::

> 	Firmware based on the |NCS| versions earlier than v2.1.0 does not support DFU.

Author: Discussed offline, leaving as-is

> The nRF5340 Audio application supports two strategies for updating the applications on the application and network core.

Reviewer: ```suggestion
The nRF5340 Audio application supports two strategies for upgrading the applications on the application and network core: single-image and multi-image.
```

> Which of the two strategies are used is controlled by setting the kconfig AUDIO_DFU to the appropriate setting.

Reviewer: ```suggestion
You control which of the two strategies is used by setting the Kconfig :kconfig:option:`CONFIG_AUDIO_DFU` to the appropriate setting.
```

> * Single-image upgrade - Uses the internal flash to upgrade either the application core or network core separately. Selected by setting :kconfig:option:`CONFIG_AUDIO_DFU` to 1.

Reviewer 2: Should we mention that this is dangerous in case the hci_rpmsg version changes?

Author: Updated warning message to sound a bit more like an encouragement for the user to investigate potential risks

> * Single-image upgrade - Uses the internal flash to upgrade either the application core or network core separately. Selected by setting :kconfig:option:`CONFIG_AUDIO_DFU` to 1.

Reviewer: We have "one sentence per line" rule in the NCS docs.
```suggestion
* Single-image upgrade - Uses the internal flash to upgrade either the application core or network core separately. 
  You can select this method by setting :kconfig:option:`CONFIG_AUDIO_DFU` to ``1``.
```

> * Multi-image upgrade - Uses external flash to upgrade both the application and network core at the same time. Selected by setting :kconfig:option:`CONFIG_AUDIO_DFU` to 2. See :ref:`multi-image DFU <ug_nrf5340_multi_image_dfu>` for more information about the process.

Reviewer: ```suggestion
* Multi-image upgrade - Uses external flash to upgrade both the application and network core at the same time. 
  You can select this method by setting :kconfig:option:`CONFIG_AUDIO_DFU` to ``2``.
  See :ref:`multi-image DFU <ug_nrf5340_multi_image_dfu>` for more information about the process.
```

Updating the SoftDevice
=======================

> Regardless of which strategy is selected for FOTA, the SoftDevice running on the network core can be upgraded.

Author: I thought the opposite was true and worked so hard to make that make sense 🥲 Will remember this for next time!

However, the current default build options for the SoftDevice creates a binary that is too large to run on the network core together with a bootloader.

> To reduce the size of the SoftDevice binary unused features in the SoftDevice can be disabled, see :ref:`softdevice_controller` documentation for more information.

Reviewer: ```suggestion
To reduce the size of the SoftDevice binary, you can disable unused features in the SoftDevice.
See :ref:`softdevice_controller` documentation for more information.
```

Entering the DFU mode
=====================

The |NCS| uses :ref:`SMP server and mcumgr <zephyr:device_mgmt>` as the DFU backend.

> The SMP server service is separate from CIS and BIS services, and is only advertised when the application is in DFU mode.

Reviewer: ```suggestion
The SMP server service is separated from CIS and BIS services, and is only advertised when the application is in the DFU mode.
```

> To enter DFU mode, **BTN 4** on the nRF5340 Audio DK must be pressed during startup.

Reviewer: ```suggestion
To enter the DFU mode, press **BTN 4** on the nRF5340 Audio DK during startup.
```

*********************

If the application is running on the nRF5340 Audio DK, you need an external flash shield to upgrade both the application and network core at the same time.
See `Requirements for external flash memory DFU`_ in the nRF5340 Audio DK Hardware documentation for more information.

.. _nrf53_audio_app_configuration_configure_fota:

Configuring FOTA upgrades
*************************

The nRF5340 Audio application supports two strategies for upgrading the applications on the application and network core: single-image and multi-image.
You control which of the two strategies is used by setting the Kconfig :kconfig:option:`CONFIG_AUDIO_DFU` to the appropriate setting.

* Single-image upgrade - Uses the internal flash to upgrade either the application core or network core separately.
  You can select this method by setting :kconfig:option:`CONFIG_AUDIO_DFU` to ``1``.
* Multi-image upgrade - Uses external flash to upgrade both the application and network core at the same time.
  You can select this method by setting :kconfig:option:`CONFIG_AUDIO_DFU` to ``2``.
  See :ref:`multi-image DFU <ug_nrf5340_multi_image_dfu>` for more information about the process.

.. caution::

>    Upgrading using the single-image upgrade strategy introduces the risk of the device being in a state where the application core and network core FW are no longer compatible, which can result in a bricked device.
>    For devices where FOTA is the only DFU method available, multi-image upgrades are recommended to ensure compatibility between the cores.
>    Please ensure you evaluate the risks for your device when selecting method.

Reviewer: ```suggestion
   Using the single-image upgrade strategy carries a risk of the device being in a state where the application core firmware and the network core firmware are no longer compatible, which can result in a bricked device.
   For devices where FOTA is the only DFU method available, multi-image upgrades are recommended to ensure compatibility between the cores.
   Make sure to evaluate the risks for your device when selecting the FOTA method.
```

Updating the SoftDevice
=======================

> You can update the SoftDevice running on the network core regardless of which strategy you select for FOTA.

Reviewer: Even simpler:
```suggestion
Both FOTA upgrade methods support updating the SoftDevice on the network core.
```

> You can update the SoftDevice running on the network core regardless of which strategy you select for FOTA.
> Regardless of which strategy is selected for FOTA, the SoftDevice running on the network core can be upgraded.

Reviewer: Previous edited sentence is still here:
```suggestion
You can update the SoftDevice running on the network core regardless of which strategy you select for FOTA.
```

Author: Missed that - removed now!

> However, the current default build options for the SoftDevice creates a binary that is too large to run on the network core together with a bootloader.

Reviewer: ```suggestion
However, the current default build options for the SoftDevice create a binary that is too large to run on the network core together with a bootloader.
```

To reduce the size of the SoftDevice binary, you can disable unused features in the SoftDevice.
See :ref:`softdevice_controller` documentation for more information.

Entering the DFU mode
=====================

The |NCS| uses :ref:`SMP server and mcumgr <zephyr:device_mgmt>` as the DFU backend.
The SMP server service is separated from CIS and BIS services, and is only advertised when the application is in the DFU mode.
To enter the DFU mode, press **BTN 4** on the nRF5340 Audio DK during startup.

> To identify the devices before the DFU takes place, the DFU mode advertising names mention the device type directly.
> The names follow the pattern in which the device *ROLE* is inserted before the ``_DFU`` suffix.

Reviewer: ```suggestion
To identify the devices before the DFU takes place, the DFU mode advertising names mention the device type directly.
The names follow the pattern in which the device role is inserted between the device name and the ``_DFU`` suffix.
```

.. _nrf53_audio_unicast_client_app_testing_steps_fota:

Testing FOTA upgrades
=====================

> To test FOTA for the nRF5340 Audio application, ensure the application is in DFU mode, and then follow the testing steps outlined in :ref:`ug_nrf53_developing_ble_fota`.

Reviewer: ```suggestion
To test FOTA for the nRF5340 Audio application, ensure the application is in the DFU mode, and then follow the testing steps outlined in :ref:`ug_nrf53_developing_ble_fota`.
```

.. _nrf53_audio_unicast_client_app_testing_steps_fota:

Testing FOTA upgrades
=====================

> To test FOTA for the nRF5340 Audio application, ensure the application is in the DFU mode, and then follow the testing steps outlined in :ref:`ug_nrf53_developing_ble_fota`.

Reviewer: ```suggestion
To test FOTA for the nRF5340 Audio application, ensure the application is in the DFU mode, and then follow the testing steps in the FOTA over Bluetooth Low Energy section of :ref:`ug_nrf53_developing_ble_fota` (you can skip the configuration steps).
```

Reviewer: @andvib , can you confirm the configuration info can be skipped?
